Life.java

```diff
@@ -42,11 +44,70 @@ public void setDead(int x, int y) {
   }
 
   public boolean isAlive(int x, int y) {
-    return raster[x][y];
+  // Verhindern von null Pointer Exception
+  if (x >= 0 && x < raster.length){
+    if (y >= 0 && y < raster[x].length){
+      return raster[x][y];
+    }
+  }
+    //Außerhalb des Rasters = tot
+    return false;
   }
 
   public ILife nextGeneration() {
-    // TODO Auto-generated method stub
-    return null;
+    Life l = new Life();
+    for (int i = 0; i< raster.length; i++){
+      for (int j = 0; j< raster[i].length; j++){
+        int neighbours = checkNeighbours(i,j);
+        if (neighbours < 2 || neighbours > 3){
+          l.setDead(i,j);
+        } else {
+          l.setAlive(i,j);
+          if (neighbours == 3){
+            l.setAlive(i,j);
+          } else {
+            boolean lastGeneration = isAlive(i,j);
+            if (lastGeneration){
+              l.setAlive(i,j);
+            } else {
+              l.setDead(i,j);
+            }
+          }
+        }
+      }
+    }
+    return l;
+  }
+
+  /*Methode zum testen ob es funktioniert/Debugging
```

Comment:
Sie können so eine Methode in ihre Klasse aufnehmen, die sogar Sinn macht. z.B. die toString() Methode:
Die sieht bei mir so aus
```
    @Override
    public String toString() {
        StringBuilder output = new StringBuilder();
        for( int x = 0; x < getDim_x(); x++ ) {
            for( int y = 0; y < getDim_y(); y++ ) {
                if( cells[ x ][ y ] == CellValue.ALIVE ) {
                    output.append( "*" );
                } else {
                    output.append( "-" );
                }
            }
            output.append( "\n" );
        }
        return output.toString();
    }
```